### PR TITLE
Select method for Field

### PR DESCRIPTION
**Pull request by lang-m**
Based on #256

### CODE REVIEW

**Comment by lang-m** [comment]
One test is currently failing. Before this can be addressed #298 has to be merged.

**Comment by lang-m** [comment]
Locally, all test for field/mesh sel pass.

File: discretisedfield/field.py [hunk 1]
```diff
@@ -2599,6 +2599,105 @@ def line(self, p1, p2, n=100):
...
+            dfu.assemble_index(
+                slice(None), self.mesh.region.ndim + 1, {dim_index: sel_index}
+            )
+        ]
+
+        try:
+            mesh = self.mesh.sel(*args, **kwargs)
+        except ValueError as e:
+            if "p1 and p2 must not be empty" not in str(e):
+                raise
```
Comment (swapneelap) [inline discretisedfield/field.py:2694]: Does this require a `raise e` or simply `raise`? I have never used this, hence I do not know.

**Review by swapneelap — APPROVED** [review]
I really like the `_sel_convert_input` method which avoids duplication in both field and mesh select methods.
I have a minute comment, however, it is more of a question than a suggestion.

File: discretisedfield/field.py (hunk 1 above)
Comment (lang-m) [reply discretisedfield/field.py:2694]: This is on purpose `raise` without an exception and re-raises the currently handled exception.

`raise e` would technically also work but then the stack trace looks a bit different (without the `e` line 2694 does not appear in the stack trace).

**Review by samjrholt — APPROVED** [review]

**Pull request merged**